content/advanced/modules/consume-example/index.md

```diff
@@ -0,0 +1,312 @@
+
+Let's see how you can consume it.
+
+Create a new YAML file:
+
+:::code{language=shell showLineNumbers=false showCopyAction=true}
+touch use-module.yaml
+:::
+
+Open this file in your chosen text editor and paste the following CloudFormation YAML:
+
+<!-- vale off -->
+:::code{language=yaml showLineNumbers=false showCopyAction=true}
+AWSTemplateFormatVersion: 2010-09-09
+
+Resources:
+
+  Vpc:
+    Type: CFNWORKSHOP::EC2::VPC::MODULE
+    Properties:
+      VpcCidr: 10.1.0.0/16
+:::
+<!-- vale on -->
+
+That's it. Nice and short isn't it? You can see why modules are going to be so useful.
+
+Let's create a new stack from that template using the following command:
+
+:::code{language=shell showLineNumbers=false showCopyAction=true}
+aws cloudformation deploy --template-file use-module.yaml --stack-name cfn-workshop-modules
+:::
+
+### Taking a closer look
+
+Now that you deployed the stack, let's take a closer look at what actually happened. It will help to understand more on how modules work in CloudFormation.
+
+Open the AWS Console and navigate to the CloudFormation service. Locate the stack that you just created and select the `Resources` tab.
+Notice that the stack is showing it has 23 resources.
+
+![stack-resources](/static/advanced/modules/StackResources.png)
+
+This number of resources can be explained if we take a look at the processed template for this stack. You can see that the actual template that CloudFormation deploys is based upon the content of the module.
+When a module is consumed in a CloudFormation template, the module resource is replaced with the resources defined for it in the module template.
+
+![stack-template](/static/advanced/modules/StackTemplate.png)
+
+
+
+
+
+### Challenge
+
+Add `Name` tags to the resources described in the module fragment, so that the resources created are easily identifiable. Users should be able to specify a value for  a new, relevant module property called `NameTag` when they consume the module.
+
+:::expand{header="Need a hint?"}
+* Update the `module.yaml` file to add `Name` tags to all the resources that support them. Documentation for adding tags to a VPC can be found [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpc.html#cfn-ec2-vpc-tags). For other resource types in the template, determine whether they support tags or not, and how: see [AWS resource and property types reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html) for more information. When done, reference the value for the `Name` tag you added to each resource supporting tags in a new `NameTag` input parameter you specify in the fragment.
+
+* Submit the changes, and set the new module version as the default version. For more information, see the `submit` [command](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-cli-submit.html) in the CloudFormation Command Line Interface reference.
```

Comment:
We should add the actual command here instead of asking them to leave the page. `cfn submit --set-default`. Unless we don't want to spoil the entire solution as a challenge?

Reply:
This is the "need a hint" section. The full command is shown a little lower down in the "Want to see the solution?" section so i think we are good.